qiskit/test/mock/utils/backend.py

```diff
@@ -13,93 +13,86 @@
 # that they have been altered from the originals.
 
 """
-Fake 100 qubit device.
+Fake backend generation.
 """
 import itertools
-import os
 import json
 from datetime import datetime
-from typing import Optional, List
+from typing import Optional, List, Type
+
 import numpy as np
 
+from qiskit.exceptions import QiskitError
 from qiskit.providers.models import (PulseBackendConfiguration,
-                                     BackendProperties, PulseDefaults, BackendConfiguration, GateConfig, UchannelLO,
-                                     Command)
+                                     BackendProperties, PulseDefaults, GateConfig, Command)
 from qiskit.providers.models.backendproperties import Nduv, Gate
 from qiskit.qobj import PulseQobjInstruction, PulseLibraryItem
 from qiskit.test.mock.fake_backend import FakeBackend
 
 
-class FakeOpenPulse100Q(FakeBackend):
-    """A fake 100Q backend."""
-
-    def __init__(self):
-        """
-        TODO: architecture
-        """
-        dirname = os.path.dirname(__file__)
-        filename = "conf_100q.json"
-        with open(os.path.join(dirname, filename), "r") as f_conf:
-            conf = json.load(f_conf)
-
-        configuration = PulseBackendConfiguration.from_dict(conf)
-        configuration.backend_name = 'fake_openpulse_100q'
-        super().__init__(configuration)
-
-    def properties(self):
-        """Returns a snapshot of device properties."""
-        dirname = os.path.dirname(__file__)
-        filename = "props_100q.json"
-        with open(os.path.join(dirname, filename), "r") as f_prop:
-            props = json.load(f_prop)
-        return BackendProperties.from_dict(props)
-
-    def defaults(self):
-        """Returns a snapshot of device defaults."""
-        dirname = os.path.dirname(__file__)
-        filename = "defs_100q.json"
-        with open(os.path.join(dirname, filename), "r") as f_defs:
-            defs = json.load(f_defs)
-        return PulseDefaults.from_dict(defs)
-
-
-# TODO: move to utils or remove if not necessary
 class FakeBackendBuilder(object):
     """FakeBackend builder.
 
     For example:
-
-        builder = FakeBackendBuilder("tashkent", grid_size=10)
+        builder = FakeBackendBuilder("Tashkent", n_qubits=100)
         path = os.path.dirname(os.path.abspath(__file__))
         builder.dump(path)
     """
 
+    _SINGLE_QUBIT_GATES = ['id', 'u1', 'u2', 'u3']
+    _DEFAULT_BASIS_GATES = ['id', 'u1', 'u2', 'u3', 'cx']
```

Comment:
This could just be in the main body of init, the standard pattern is
```python
if basis_gates is None:
    basis_gates = ['id', 'u1', 'u2', 'u3', 'cx']
```
and then document the default in the docstring.